crates/typst-cli/src/args.rs

```diff
@@ -713,14 +795,27 @@ fn parse_page_number(value: &str) -> Result<NonZeroUsize, &'static str> {
 }
 
 /// The clap value parser used by `SharedArgs.input`
-fn input_value_parser() -> impl TypedValueParser<Value = Input> {
+fn file_input_value_parser() -> impl TypedValueParser<Value = FileInput> {
+    clap::builder::OsStringValueParser::new().try_map(|value| {
+        if value.is_empty() {
+            Err(clap::Error::new(clap::error::ErrorKind::InvalidValue))
+        } else if value == "-" {
+            Ok(FileInput::Stdin)
+        } else {
+            Ok(FileInput::Path(value.into()))
+        }
+    })
+}
+
+/// The clap value parser used by `EvalCommand.statement`
+fn string_input_value_parser() -> impl TypedValueParser<Value = StringInput> {
     clap::builder::OsStringValueParser::new().try_map(|value| {
         if value.is_empty() {
             Err(clap::Error::new(clap::error::ErrorKind::InvalidValue))
         } else if value == "-" {
-            Ok(Input::Stdin)
+            Ok(StringInput::Stdin)
         } else {
-            Ok(Input::Path(value.into()))
+            Ok(StringInput::String(value.to_string_lossy().into()))
```

Comment:
I'm not sure we want to be lossy here. Is this the default behavior of clap when a string is required or does it error?